#161342842 Rate an article

victor-nweze-deel opened the pull request:
#### What does this PR do?
Rate an article

#### Description of Task to be completed?
- Create a new migration and model for ratings
- Create a route handler for posting new ratings
- Write unit and integration tests for newly added scripts
- Write more tests to improve coverage

#### How should this be manually tested?
- Git clone this repository (Run `git clone https://github.com/andela/haven-ah-backend.git`)
- Navigate into the haven-ah-backend folder (Run `cd haven-ah-backend`)
- Start postman locally
- Start the development server (Run `npm run start-dev`)
- Fire up the route `localhost:5000/api/v1/articles/:slug/rating` on postman with a rating
- To run the unit and integration test run `npm test` on you terminal

#### What are the relevant pivotal tracker stories?
[#161342842](https://www.pivotaltracker.com/story/show/161342842)

#### Screenshots (if appropriate)
<img width="1215" alt="screen shot 2018-11-06 at 7 57 55 pm" src="https://user-images.githubusercontent.com/41582565/48116588-d7cedc80-e266-11e8-976f-d0d1244cbf37.png">
<img width="1209" alt="screen shot 2018-11-06 at 7 58 12 pm" src="https://user-images.githubusercontent.com/41582565/48116590-d9000980-e266-11e8-867b-44385a1e52e3.png">

PriscillaSam commented:
This is well-done @ProfJigsaw. 👍

ucheg6 commented:
This looks great. good job victor

TheoOkafor reviewed (commented):
Looks good. 👍

seyi-adeleke commented on server/routes/article.js:
```diff
@@ -11,6 +11,12 @@ router.post(
 );
+router.post(
+  '/articles/:slug/rating',
+  isAuthenticated,
```
Can a user rate their own/article?

victor-nweze-deel replied:
Yes they can, but one rating per article like other readers.

seyi-adeleke commented on server/tests/repositoryTests/ratings.test.js:
```diff
@@ -0,0 +1,10 @@
+
+describe('Ratings:', () => {
+  it('should be a function', () => {
+    expect(ratingRepo.createRating).to.be.a('function');
```
This test doesn't  tell me anything

victor-nweze-deel replied:
I have more tests on the `rating` functionality in the article.test.js file since ratings pertain to articles. I created this to have at least one test on the rating repo

victor-nweze-deel committed:
ft(ratings): Rate an article

- Create a new migration and model for ratings
- Create a route handler for posting new ratings
- Write unit and integration tests for newly added scripts
- Write more tests to improve coverage
- Update article repository function
